lib/MusicBrainz/Server/WebService/Serializer/JSON/LD/Artist.pm

```diff
@@ -41,7 +41,18 @@ around serialize => sub {
         if (@members) {
             my %seen_members;
             for my $member (@members) {
-                $seen_members{$member->target->gid} = member_relationship($member, $inc, $stash, $seen_members{$member->target->gid});
+                # We separate different dates, but keep together
+                # different roles with the same dates
+                my $key = $member->target->gid;
+                my $begin_date = $member->link->begin_date;
+                my $end_date = $member->link->end_date;
+                if ($begin_date && $begin_date->defined_run) {
+                    $key .= '-' . join('-', $begin_date->defined_run);
+                }
+                if ($end_date && $end_date->defined_run) {
+                    $key .= '-' . join('-', $end_date->defined_run);
+                }
```

Comment:
if one rel has begin_date: 1999 and another has end_date: 1999 wouldn't they end up with the same key here? maybe we should prefix the key parts with begin/end.

Reply:
Oh, damn, yes, good point. Will do :)